### PR TITLE
feat: simple rotation

### PR DESCRIPTION
**Pull request by feugy**
### What's in there?

Closes https://github.com/pinojs/pino/issues/1323

This module provides a basic Pino transport to write in log files.
Files are automatically rolled based on frequency and/or size.

Supported options:
- `frequency`: either 'daily', 'hourly', or a number of ms
- `size`:  a string with a number and a unit ('g', 'm', 'k'), or a number of MB
- `file`: path to the log folder and file. Numbers will be appended to the final files. 
- `extension`: custom string you can append after the file number
- `mkdir`: automatically creates missing folders

### How to test?

Here is a very simple snippet to write in log every 100ms, and rotate them every 10s:
```js
// file logger.js
// run with `node logger`

const { join } = require('path')
const pino = require('pino')

const transport = pino.transport({
  target: '.',
  options: { file: join('logs', 'log.json'), frequency: 10e3, mkdir: true }
})

const logger = pino(transport)

let i = 0
setInterval(() => logger.info(`frame #${i++}`), 100)
```

Then you can play with the various options.

### CODE REVIEW

File: pino-roll.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+  function roll () {
+    destination.reopen(buildFileName(file, ++number, extension))
+  }
+
+  function scheduleRoll () {
+    clearTimeout(rollTimeout)
+    rollTimeout = setTimeout(() => {
+      roll()
+      frequencySpec = parseFrequency(frequency)
```
Comment (mcollina) [inline pino-roll.js:84]: Can we avoid parsing again?

File: pino-roll.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  if (frequencySpec) {
+    destination.once('close', () => {
+      clearTimeout(rollTimeout)
+    })
+    scheduleRoll()
+  }
+
+  if (maxSize) {
+    // patching sonic-boom stream is fragile, but it's the most performant way to "watch" the file's size
+    const originalRelease = destination.release.bind(destination)
```
Comment (mcollina) [inline pino-roll.js:61]: I would really _not_ recommend anybody to do this. `destination.release` is one of the most tricky methods of sonic-boom. Can we add a feature there to simplify this?

**Comment by simoneb** [comment]
@feugy what plans do you have for this? If you're not planning to continue working on this can you add me to the repo so I can take over?

**Comment by feugy** [comment]
Hi @simoneb. I was distracted recently, but I'd like to keep working on this until completion.

**Comment by simoneb** [comment]
Sounds good @feugy, thanks! If at any point you need any help or input, shout!

File: pino-roll.js (hunk 2 above)
Comment (feugy) [reply pino-roll.js:61]: https://github.com/pinojs/sonic-boom/pull/140 is an attempt to make this information available without forcing use to monkey-patch SonicBoom.

**Comment by DRoet** [comment]
Can we expect to see a release soon now that the `sonic-boom` changes have been merged and released? looking forward to try this out in Fastify 4.x

**Comment by feugy** [comment]
Hello @DRoet!
Absolutely. I have some bandwidth tomorrow, so I will iterate on that PR, and probably publish a beta version on NPM.

**Comment by simoneb** [comment]
Shout if you need any support here

**Pull request merged**